### PR TITLE
Initial content of project

### PR DESCRIPTION
**Pull request by illicitonion**
This is adapted from
https://syllabus.codeyourfuture.io/js-core-3/tv-show-dom-project/ with roughly the following significant changes:
1. We alternate partners between repos, so people get practice working on other people's code. This also entails an "understand/discuss/compare approaches" pieces each level, and a "do some refactoring" piece each level. Note that we use a template repo, rather than forks, so that people can fork their partner's repo (otherwise GitHub will prevent them, because both forks will be of the same upstream, and GitHub doesn't like that).
2. Levels 200 and 300 (episode search + drop-down) are merged into level 200. This is so that they're completed by one person before swapping repos.
3. Level 300 (ex-350) requires deleting `episodes.js`. And we don't ever include `shows.js`, instead telling people to use fetch.
4. Level 300 (ex-350) explicitly states that there should be a "loading" screen, and that errors should be reported to the user.
5. We have stricter explicit requirements: Never fetch the same URL twice (rather than more vague "avoid repeated fetches" guidance).
6. Accessibility and semantic HTML are requirements rather than extras.

### CODE REVIEW

File: levels/level-400.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+Until now, your site has only showed information about the episode of one TV show.
+
+But TVmaze has information about lots of TV shows, all in the same format.
+
+We want to allow displaying any of them.
+
+### Requirements
+
+1. Add a `select` element to your page which allows the user to choose which show they want to see the episodes of.
```
Comment (SallyMcGrath) [inline levels/level-400.md:35]:
```suggestion
1. Add a `select` element to your page so the user can choose a show.
```

File: levels/level-400.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+2. When the user first loads the page, you should make a `fetch` request to https://api.tvmaze.com/shows ([documentation](https://www.tvmaze.com/api#show-index)) to get a list of available shows, and add an entry to the drop-down per show.
```
Comment (SallyMcGrath) [inline levels/level-400.md:36]:
```suggestion
2. When the user first loads the page, make a `fetch` request to https://api.tvmaze.com/shows ([documentation](https://www.tvmaze.com/api#show-index)) to get a list of available shows, and add an entry to the drop-down per show.
```

File: levels/level-400.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+3. When a user selects a show, your app should display the episodes for that show as per the earlier levels of this project.
```
Comment (SallyMcGrath) [inline levels/level-400.md:37]:
```suggestion
3. When a user selects a show, display the episodes for that show, just like the earlier levels of this project.
```

File: levels/level-400.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+  Note that you will need to perform a `fetch` to get the episode list.
```
Comment (SallyMcGrath) [inline levels/level-400.md:39]:
```suggestion
You will need to perform a `fetch` to get the episode list.
```

File: levels/level-400.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+4. Ensure that your search and episode selector controls still work correctly when you switch shows.
```
Comment (SallyMcGrath) [inline levels/level-400.md:40]:
```suggestion
4. Make sure that your search and episode selector controls still work correctly when you change shows.
```

File: levels/level-400.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+5. This show select must list shows in alphabetical order, case-insensitive.
```
Comment (SallyMcGrath) [inline levels/level-400.md:41]:
```suggestion
5. Your select must list shows in alphabetical order, case-insensitive.
```

File: readme.md [hunk 7]
```diff
@@ -1,7 +1,66 @@
...
+        }
+    }
+}
+```
+
+## Rules
+
+### Your website must be accessible and use semantic HTML
+
+TODO: How exactly should we set requirements here?
```
Comment (SallyMcGrath) [inline readme.md:42]:
```suggestion
As always, your application must score 100 for Accessibility on Lighthouse

```

File: readme.md [hunk 8]
```diff
@@ -1,7 +1,66 @@
...
+
+### Don't change `getAllEpisodes()`
+
+Because we're just using this `getAllEpisodes()` function as a placeholder until we use `fetch`, and we can't change what data will be returned to us from the `fetch`, you shouldn't change the implementation of `getAllEpisodes()`.
+
+### Use tech that we've learnt
+
+You should use the DOM-manipulation functions you've been learning about on the course - the point of this project is to practice.
+
+You should not use React, Vue, JQuery, or any other libraries or frameworks for this project.
```
Comment (SallyMcGrath) [inline readme.md:52]:
```suggestion
You should not use React, Vue, JQuery, or any other libraries or frameworks for this project. You can rebuild it in React later. Or maybe in Svelte!

```

File: readme.md [hunk 9]
```diff
@@ -1,7 +1,66 @@
...
+
+## Get started
+
+Each level has a .md file in the [`levels`](./levels) folder. Start with level 0, then complete level 100.
+
+After that, you're going to swap codebases with someone else. You should complete level 200 in their repository, not your own. Every level, you'll swap back - so you'll do level 300 in your own codebase, level 400 in theirs, etc.
+
+You are allowed to change anything in your partner's code to make it easier to understand or build on, but they must understand and accept your changes.
+
+Remember, finish each level before you move on to the next. Reach out for help if you're stuck.
```
Comment (SallyMcGrath) [inline readme.md:66]:
```suggestion
Remember, finish each level before you move on to the next. Reach out for help when you're stuck.

```

File: readme.md [hunk 10]
```diff
@@ -1,7 +1,66 @@
...
+
+### Use tech that we've learnt
+
+You should use the DOM-manipulation functions you've been learning about on the course - the point of this project is to practice.
+
+You should not use React, Vue, JQuery, or any other libraries or frameworks for this project.
+
+### Functionality before style
+
+Your goal of this project is to achieve the levels, all of which are about functionality. When you have completed level 500, you can spend some time making it look beautiful. But focus on the functionality first.
```
Comment (SallyMcGrath) [inline readme.md:56]:
```suggestion
Your goal of this project is to achieve the levels, all of which are about functionality. When you have completed level 500, you can spend some time making it look beautiful. But focus on the functionality first.

Remember: it's important that software works and that people can use it.
```

**Review by SallyMcGrath — APPROVED** [review]
Just done a minor edit -- I will look again tomorrow. Good Hemingway scores!

File: levels/level-200.md [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+
+Compare their implementation to yours. Think:
+1. How is it different?
+2. What do you prefer about your implementation?
+3. What do you prefer about their implementation?
+4. What did you learn that you didn't know before?
+
+They should do the same with your repository.
+
+Have a discussion about your answers to these questions. In class, together you should give a 3 minute talk about your conclusions.
```
[the remark this answers is not in the document]
Comment (illicitonion) [reply levels/level-200.md:21]: I gave them five minutes, for this format :)

File: levels/level-300.md [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+
+## Adding new functionality
+
+Level 300 is about getting rid of our static data from `episodes.js`, and instead using an API.
+
+### Requirements
+
+1. You must delete the `episodes.js` file from your repository.
+2. Your website must still continue to work the same, but by using a `fetch` request to https://api.tvmaze.com/shows/82/episodes (note: this URL should serve the exact same content as was returned by `getAllEpisodes` in `episodes.js`).
+3. You must only fetch this URL once per visit to your website. You should not re-fetch when someone searches, scrolls, or selects an episode from the drop-down.
```
[the remark this answers is not in the document]
Comment (illicitonion) [reply levels/level-300.md:29]: I moved the only to before the once, but kept it.

**Pull request merged**